SwiftyBluetooth/Source/CentralProxy.swift

```diff
@@ -144,7 +144,7 @@ extension CentralProxy {
         let weakRequest = timer.userInfo as! Weak<PeripheralScanRequest>
         
         if weakRequest.value != nil {
-            self.stopScan()
```

Comment:
I see the way it is set up, but it's inconsistent with the `.connect` and `.disconnect` methods throwing an error on timeout (`request.invokeCallbacks(result: .failure(SBError.operationTimedOut(operation: .connectPeripheral)))` and `        request.invokeCallbacks(result: .failure(SBError.operationTimedOut(operation: .disconnectPeripheral)))`) and also makes it difficult to distinguish between when the scanner stops due to the timer running our vs. the user calling `.stopScan()` once they have succesfully found their peripheral of interest. It makes much less sense to me to have the user have to pass a custom error to `.stopScan()` in order to ensure that there is not confusion between stopping the scan when successful and when unsuccessful. How would you suggest otherwise achieving this functionality with the library as-is?